contracts/actions/mcd/McdPayback.sol

```diff
@@ -76,25 +73,65 @@ contract McdPayback is ActionBase, McdHelper {
     function _mcdPayback(Params memory _inputData) internal {
         IManager mcdManager = IManager(_inputData.mcdManager);
 
-        address urn = mcdManager.urns(_inputData.vaultId);
-        bytes32 ilk = mcdManager.ilks(_inputData.vaultId);
+        (address urn, bytes32 ilk) = getUrnAndIlk(_inputData.mcdManager, _inputData.vaultId);
 
         // if _amount is higher than current debt, repay all debt
         uint256 debt = getAllDebt(address(vat), urn, urn, ilk);
         _inputData.amount = _inputData.amount > debt ? debt : _inputData.amount;
         // pull Dai from user and join the maker pool
         DAI_ADDR.pullTokensIfNeeded(_inputData.from, _inputData.amount);
         DAI_ADDR.approveToken(DAI_JOIN_ADDR, _inputData.amount);
-        IDaiJoin(DAI_JOIN_ADDR).join(urn, _inputData.amount);
-        // decrease the vault debt
-        mcdManager.frob(_inputData.vaultId, 0, normalizePaybackAmount(address(vat), urn, ilk));
-
-        logger.Log(
-            address(this),
-            msg.sender,
-            "McdPayback",
-            abi.encode(_inputData, debt)
+
+        if (_inputData.mcdManager == CROPPER) {
+            address owner = ICdpRegistry(CDP_REGISTRY).owns(_inputData.vaultId);
```

Comment:
any reason lines 86/88 aren't in _cropperPayback, and line 92 in _mcdManagerPayback?